docusaurus/docs/dev-docs/backend-customization.md

```diff
@@ -4,30 +4,67 @@ description: All elements of Strapi's back end, like routes, policies, middlewar
 
-- the [controllers](/dev-docs/backend-customization/controllers) that execute code once a route has been reached,
+A request can travel through the Strapi backend as follows:
 
-- the [services](/dev-docs/backend-customization/services) that are used to build custom logic reusable by controllers,
+1. The Strapi server receives a [request](/dev-docs/backend-customization/requests-responses#requests).
+2. The request hits [global middlewares](/dev-docs/backend-customization/middlewares) that are run in a sequential order.
+3. The request hits a [route](/dev-docs/backend-customization/routes).<br/>By default, Strapi generates routes for all the content-types that you create (see [REST API documentation](/dev-docs/api/rest)), and more routes can be added and configured.
+4. [Route policies](/dev-docs/backend-customization/policies) act as a read-only validation step that can block access to a route. [Route middlewares](/dev-docs/backend-customization/routes#middlewares) can control the request flow and mutate the request itself before moving forward.
+5. [Controllers](/dev-docs/backend-customization/controllers) execute code once a route has been reached. [Services](/dev-docs/backend-customization/services) are optional, additional code that can be used to build custom logic reusable by controllers.
+6. The code executed by the controllers and services interacts with the [models](/dev-docs/backend-customization/models) that are a representation of the content data structure stored in the database.<br />Interacting with the data represented by the models is handled by the [Entity Service](/dev-docs/api/entity-service) and [Query Engine](/dev-docs/api/query-engine).
+7. The server returns a [response](/dev-docs/backend-customization/requests-responses#response). The response can travel back through route middlewares and global middlewares before being sent.
 
-- the [models](/dev-docs/backend-customization/models) that are a representation of the content data structure,
+Both global and route middlewares include an asynchronous callback function, `await next()`. Depending on what is returned by the middleware, the request will either go through a shorter or longer path through the backend:
 
-- the [responses](/dev-docs/backend-customization/requests-responses#responses) sent to the application that sent the request,
+* If a middleware returns nothing, the request will continue travelling through the various core elements of the backend (i.e., controllers, services, and the other layers that interact with the database).
+* If a middleware returns something, a response will be immediately sent, skipping the rest of the core elements.
 
-- and the [webhooks](/dev-docs/backend-customization/webhooks) that are used to notify other applications of events that occurred.
+```mermaid
+graph TB
```

Comment:
This graph is incorrect.

on every layer you except for policy's you could return before calling the next faction inline.
so for example giving a policy error means that global middleware still get ran.
Or if you where to return before calling query engine in entity service it will take the same route down as it came up.
So if you return before calling next with the 4th decorator decorator 3s next would be called and then go back down the chain.
Will make a pr on top of this one with a more accurate route.

Reply:
cc @derrickmehaffy since the content of this graph comes from some material Derrick sent me and that was presented during a best practices showcase.